topi/python/topi/cuda/vision.py

```diff
@@ -0,0 +1,160 @@
+import tvm
+from .. import tag
+from .. import generic
+
+@generic.schedule_reorg.register(["cuda", "gpu"])
+def schedule_reorg(outs):
+    """Schedule for reorg operator.
+
+    Parameters
+    ----------
+    outs: Array of Tensor
+        The computation graph description of reorg
+        in the format of an array of tensors.
+
+    Returns
+    -------
+    s: Schedule
+        The computation schedule for reorg.
+    """
+    target = tvm.target.current_target()
+    if target.target_name == "cuda" and "cublas" in target.libs:
+        return generic.schedule_extern(outs)
+    outs = [outs] if isinstance(outs, tvm.tensor.Tensor) else outs
+    s = tvm.create_schedule([x.op for x in outs])
+    def _schedule_reorg(Reorg):
+        num_thread = 64#tvm.target.current_target(allow_none=False).max_num_threads
+        bx, tx = s[Reorg].split(Reorg.op.axis[0], factor=num_thread)
+        s[Reorg].bind(bx, tvm.thread_axis("blockIdx.x"))
+        s[Reorg].bind(tx, tvm.thread_axis((0, num_thread), "threadIdx.x"))
+        Out = outs[0].op.output(0)
+        s[Reorg].compute_at(s[Out], s[Out].op.axis[1])
+        tx, xi = s[Out].split(Out.op.axis[0], nparts=num_thread)
+        s[Out].bind(tx, tvm.thread_axis((0, num_thread), "threadIdx.x"))
+
+    def _traverse(OP):
+        # inline all one-to-one-mapping operators except the last stage (output)
+        if tag.is_injective(OP.tag):
+            if OP not in s.outputs:
+                s[OP].compute_inline()
+            for tensor in OP.input_tensors:
+                if tensor.op.input_tensors:
+                    _traverse(tensor.op)
+        # schedule reorg
+        elif OP.tag == 'reorg':
+            Reorg = OP.output(0)
+            _schedule_reorg(Reorg)
+        else:
+            raise RuntimeError("Unsupported operator: %s" % OP.tag)
+    _traverse(outs[0].op)
+    return s
+
+@generic.schedule_shortcut.register(["cuda", "gpu"])
+def schedule_shortcut(outs):
+    """Schedule for shortcut operator.
+
+    Parameters
+    ----------
+    outs: Array of Tensor
+        The computation graph description of shortcut
+        in the format of an array of tensors.
+
+    Returns
+    -------
+    s: Schedule
+        The computation schedule for shortcut.
+    """
+
+    outs = [outs] if isinstance(outs, tvm.tensor.Tensor) else outs
+    s = tvm.create_schedule([x.op for x in outs])
+
+    def _schedule_shortcut(Shortcut):
+        num_thread = 64#tvm.target.current_target(allow_none=False).max_num_threads
+        if Shortcut.op in s.outputs:
+            Out = Shortcut
+        else:
+            Out = outs[0].op.output(0)
+            s[Shortcut].compute_at(s[Out], s[Out].op.axis[1])
+        k = Out.op.axis[0]
+        bx, tx = s[Out].split(k, factor=num_thread)
+        s[Out].bind(bx, tvm.thread_axis("blockIdx.x"))
+        s[Out].bind(tx, tvm.thread_axis("threadIdx.x"))
+
+    def _traverse(OP):
+        print("Traverse OPs:", OP, OP.tag, OP.input_tensors)
```

Comment:
Does this mean we can simply use topi.cuda.schedule_injective?

Reply:
i tried this, but its giving some trouble..

/media/siju/DataDisk/work/working_copy/tvm/apps/android_rpc/app/src/main/jni/../../../../../../include/../src/runtime/module_util.cc:46: Check failed: ret == 0 (-1 vs. 0) [08:27:45] /media/siju/DataDisk/work/working_copy/tvm/apps/android_rpc/app/src/main/jni/../../../../../../include/../src/runtime/opencl/opencl_module.cc:210: Check failed: e == CL_SUCCESS OpenCL Error, code=-38: CL_INVALID_MEM_OBJECT

Reply:
what is the difference, have you take a look at the code? Could it due to the maximum number of thread difference?

Reply:
In that case, we can try to set the number of threads differently